sklearn/impute/_iterative.py

```diff
@@ -685,23 +687,33 @@ def _validate_limit(limit, limit_type, n_features):
             Type of limit to validate.
         n_features: int
             Number of features in the dataset.
+        is_empty_feature: ndarray, shape (n_features, )
+            Mask array indicating empty feature imputer has seen during fit.
+        keep_empty_feature: bool
+            If False, remove empty-feature indices from the limit.
 
         Returns
         -------
         limit: ndarray, shape(n_features,)
             Array of limits, one for each feature.
         """
+        n_features_in = len(is_empty_feature)
+        if limit is not None and not np.isscalar(limit) and len(limit) != n_features_in:
```

Comment:
since we're not validating them, I think `_num_samples(limit)` makes more sense here.